system-helper/flatpak-system-helper.c

```diff
@@ -141,6 +205,123 @@ no_progress_cb (OstreeAsyncProgress *progress, gpointer user_data)
 {
 }
 
+static gboolean
+handle_create_pull_cache_dir (FlatpakSystemHelper   *object,
+                              GDBusMethodInvocation *invocation,
+                              const gchar           *arg_ref,
+                              const gchar           *optional_commit,
+                              const gchar           *arg_installation)
+{
+  GDBusConnection *connection = g_dbus_method_invocation_get_connection (invocation);
+  const gchar *sender = g_dbus_method_invocation_get_sender (invocation);
+  FlatpakSystemHelperOngoingPull *new_pull;
+  g_autoptr(FlatpakDir) system = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autoptr(GDBusMessage) msg = NULL;
+  g_autoptr(GDBusMessage) reply = NULL;
+  g_autofree gchar *cache_tmpdir = NULL;
+  g_autofree gchar *user_folder = NULL;
+  g_autofree gchar *root_folder = NULL;
+  g_autofree gchar *flatpak_dir = NULL;
+  guint32 uid;
+  guint uid_pull_count = 0;
+
+  system = dir_get_system (arg_installation, &error);
+  if (system == NULL)
+    {
+      g_dbus_method_invocation_return_gerror (invocation, error);
+      return TRUE;
+    }
+
+  if (ongoing_pulls == NULL)
+    ongoing_pulls = g_ptr_array_new ();
+
+  msg = g_dbus_message_new_method_call ("org.freedesktop.DBus",
+                                        "/org/freedesktop/DBus",
+                                        "org.freedesktop.DBus",
+                                        "GetConnectionCredentials");
+  g_dbus_message_set_body (msg, g_variant_new ("(s)", sender));
+
+  reply = g_dbus_connection_send_message_with_reply_sync (connection, msg,
```

Comment:
It would be significantly simpler to call `g_dbus_connection_call_sync()` instead.